lessons/09-clientside-frameworks/INCLASS.md

```diff
@@ -0,0 +1,65 @@
+- [Angular Data Binding](https://docs.angularjs.org/guide/databinding)
+
+##JSX
+In essence, JSX is a preprocessor step that adds XML syntax to JavaScript. One can definitely use React without JSX but JSX makes React a lot more elegant. JSX tags have a tag name, attributes and nested children. In addition to the above, XML has the benefit of balanced opening and closing tags. This helps make large trees easier to read than function calls or object literals. For more information on JSX, visit [here](https://facebook.github.io/react/docs/jsx-in-depth.html).
+
+##Which one to use?
+Both frameworks can be complementary to each other. If you are choosing to build a web application, it is important that you leverage new tooling, which can increase the quality of your brand interactions and decrease development time by leveraging existing libraries or frameworks. If your application has a lot of large data operations and dynamic content, **React** is an excellent addition to the stack. The downside to using both frameworks is that it can increase complexity within the application.
+
+##Hello Angular 1.0
+In it's simplest form, one can use the code provided below to asynchronously update and place the content from the textbox to complete the period next to "Hello".
+```html
+<!DOCTYPE html>
+<html ng-app>
+<head>
+    <title>Hello World, AngularJS</title>
+    <script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/angularjs/1.0.7/angular.min.js"></script>
+</head>
+<body>
+    
+    Write some text in textbox:
+    <input type="text" ng-model="sometext" />
+ 
+    <h1 ng-show="sometext">Hello {{ sometext }}</h1>
+     
+</body>
+</html>
+```
+The first thing we notice is the use of an attribute **ng-app** with the html tag. This attribute, known as a directive, signals Angular to be active in this portion of the page. Another important thing to note is the use of the **ng-model** directive. **ng-model** binds the state of the textbox with the model value, i.e. "sometext". As a result, whenever the textbox value changes, Angular authomatically changes the model "sometext" to the respective value. This is what is known as Two-way Data Binding. Moreover, **{{sometext}}** tells Angular to bind the value of the model "sometext" in its place. Finally, the **ng-show** directive conditionally shows an element, depending on the value of the relative boolean expression. It's default state is False, and one could also use **ng-hide**, which does exactly the opposite of **ng-show**.
+
+Angular provides an interesting mechanism one can use to filter the bound information prior to presentation. One could also substitute the h1 tag above with the following, where "sometext" represents the expression and "uppercase" or "lowercase" the targetted filter.  
+
+```html
+<h1>Uppercase: {{ sometext | uppercase }}</h1>
+<h1>Lowercase: {{ sometext | lowercase }}</h1>
+```
+
+##Hello Angular 2.0
```

Comment:
This naming convention may be a bit confusing given that Angular 2.0 is an actual version of angular which could be interpreted as you introducing the various versions.